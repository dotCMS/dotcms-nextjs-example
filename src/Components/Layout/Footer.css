footer {
    background-color: var(--black);
    padding: 7rem 0;
}

footer p {
    margin: 0;
}