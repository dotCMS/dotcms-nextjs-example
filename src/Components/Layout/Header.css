header.header__menu {
    display: block;
    z-index: 1;
}

header.header__menu > button {
    float: right;
}

.header{
    background-color:black;
    margin-bottom:0px;
    padding-bottom:0px;
    z-index: 999;
}

.navbar-nav{
    justify-content:flex-end;
}


.navbar-brand {
    text-align: left;
    color: var(--white);
    font-size: 1.25rem;
    font-weight: 700;
    text-transform: uppercase;
}


.header__breadcrumbs {
    color: var(--white);
    position: relative;
}

.layout__content {
    margin-top:0px!important;


}
