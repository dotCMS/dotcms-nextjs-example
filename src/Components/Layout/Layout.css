.layout__content {
    margin-top: calc(var(--header-height) - 20px);
}

.layout__content--no-header {
    margin-top: 0;
}
