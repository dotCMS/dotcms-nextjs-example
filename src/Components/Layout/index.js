export * from './Layout.js';
export * from './LayoutWithSidebar.js';
export * from './Sidebar.js';