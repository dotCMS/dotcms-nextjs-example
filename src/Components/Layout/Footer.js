import React from 'react';
import { Container as BootstrapContainer } from 'reactstrap';

import './Footer.css';

const Footer = (props) => {
    return (
        <footer>
            <BootstrapContainer>
                <p className="text-center">DotCMS SPA Starter Site &copy;2019 &nbsp; | &nbsp; <a href="/login/profile">My Profile</a></p>
            </BootstrapContainer>
        </footer>
    );
};

export default Footer;
