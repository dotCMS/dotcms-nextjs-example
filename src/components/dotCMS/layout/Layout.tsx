// Dependencies
import * as React from 'react'

// Internals
import { PageContext } from '@/contexts'
import { Nav } from './Nav'
import Footer from './Footer'

export const Layout: React.FC = ({ children }) => {
  const {
    // @ts-ignore TODO: add the correct types
    pageRender: {
      layout: { header, footer },
    },
  } = React.useContext(PageContext)

  return (
    <>
      {header ? <Nav /> : null}
      {children}
      {footer ? <Footer /> : null}
    </>
  )
}

export default Layout
