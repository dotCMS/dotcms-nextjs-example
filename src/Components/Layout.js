import React from 'react';
import { Container as BootstrapContainer } from 'reactstrap';
import Header from './Header';

const Layout = (props) => {
    return (
        <>
            <Header title={props.title} />
            <BootstrapContainer className="layout__content">{props.children}</BootstrapContainer>
        </>
    );
};

export default Layout;
