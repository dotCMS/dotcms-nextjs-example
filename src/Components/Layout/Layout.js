import React from 'react';
import { Container as BootstrapContainer } from 'reactstrap';

import Header from './Header';
import Footer from './Footer';
import LayoutWithSidebar from './LayoutWithSidebar';

import './Layout.css';

export const Layout = (props) => {
    const contentClass = `layout__content ${!props.header && 'layout__content--no-header'}`;
    const { sidebar } = props;

    return (
        <>
            {props.header ? <Header title={props.title} /> : null}
            <BootstrapContainer className={contentClass}>
                {sidebar && !!sidebar.location ? (
                    <LayoutWithSidebar sidebar={sidebar}>{props.children}</LayoutWithSidebar>
                ) : (
                    props.children
                )}
            </BootstrapContainer>
            {props.footer ? <Footer /> : null}
        </>
    );
};
