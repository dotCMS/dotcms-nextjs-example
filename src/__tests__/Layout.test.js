import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Layout from '../Components/Layout';

describe('<Layout />', () => {
    it('should render  correctly', () => {
        const wrapper = shallow(
            <Layout>
                <p>Hola Mundo</p>
            </Layout>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});