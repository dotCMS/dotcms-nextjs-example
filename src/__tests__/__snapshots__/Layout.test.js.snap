// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render  correctly 1`] = `
<Fragment>
  <Header />
  <Container
    className="layout__content"
    tag="div"
  >
    <p>
      Hola Mundo
    </p>
  </Container>
</Fragment>
`;
